src/main/java/uk/ac/ebi/subs/validator/filereference/FileReferenceHandler.java

```diff
@@ -0,0 +1,42 @@
+package uk.ac.ebi.subs.validator.filereference;
+
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import uk.ac.ebi.subs.validator.core.validators.ValidatorHelper;
+import uk.ac.ebi.subs.validator.data.AssayDataValidationMessageEnvelope;
+import uk.ac.ebi.subs.validator.data.SingleValidationResult;
+import uk.ac.ebi.subs.validator.data.SingleValidationResultsEnvelope;
+import uk.ac.ebi.subs.validator.data.structures.SingleValidationResultStatus;
+import uk.ac.ebi.subs.validator.data.structures.ValidationAuthor;
+import uk.ac.ebi.subs.validator.util.ValidationHelper;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static uk.ac.ebi.subs.validator.util.ValidationHelper.generateSingleValidationResultsEnvelope;
+
+@Service
+@RequiredArgsConstructor
+public class FileReferenceHandler {
+
+    @NonNull
+    private FileReferenceValidator fileReferenceValidator;
+
+    public SingleValidationResultsEnvelope handleValidationRequest(AssayDataValidationMessageEnvelope envelope) {
+        String submissionId = envelope.getSubmissionId();
+
+        List<SingleValidationResult> interestingResults = fileReferenceValidator.validate(submissionId).stream()
+                .filter(ValidationHelper::statusIsNotPassOrPending)
```

Comment:
@ke4 I'm not sure why are we filtering the single validation results here 🤔can you explain the reasoning behind it? The method is called `handleValidationRequest` but then it's just creating a validation result, where is the validation happening?

Reply:
This is more or less the same code as in https://github.com/EMBL-EBI-SUBS/validation-service/blob/4aef2aaab10ddeab0376cb28d6e6d44a645b653a/src/main/java/uk/ac/ebi/subs/validator/core/handlers/AbstractHandler.java#L21
We could have a walkthrough on this process tomorrow morning.